src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12939,9 +12939,19 @@ private bool IsPossibleCollectionElement()
 
         private CollectionElementSyntax ParseCollectionElement()
         {
-            return IsAtDotDotToken()
-                ? _syntaxFactory.SpreadElement(this.EatDotDotToken(), this.ParseExpressionCore())
-                : _syntaxFactory.ExpressionElement(this.ParseExpressionCore());
+            if (this.CurrentToken.ContextualKind == SyntaxKind.WithKeyword &&
+                this.PeekToken(1).Kind == SyntaxKind.OpenParenToken &&
+                IsFeatureEnabled(MessageID.IDS_FeatureCollectionExpressionArguments))
```

Comment:
It would be good if we can make a LangVersion error work in binding, if we can't bind `with` in `[with()]`, to hint that maybe you meant to update your language version instead. There may be a similar error for `record Rec() { ... }` in language versions where `record` is treated as a method return type, or some similar case like that.